db/blip_handler.go

```diff
@@ -514,7 +514,17 @@ func (bh *blipHandler) handleChanges(rq *blip.Message) error {
 
 		}
 
-		if missing == nil && deletedFlags&(changesDeletedFlagRevoked|changesDeletedFlagRemoved) == 0 {
+		if bh.blipContext.ActiveProtocol() == BlipCBMobileReplicationV3 && deletedFlags&(changesDeletedFlagRevoked|changesDeletedFlagRemoved) != 0 && deletedFlags&changesDeletedFlagDeleted == 0 {
```

Comment:
Need to do this only for purge on removal?

Reply:
Do we actually need to exempt tombstones from purge on revoked/removal?

Reply:
Changed to only do on purge on removal and removed deleted checks.